Sources/Workspace/Workspace.swift

```diff
@@ -89,6 +89,38 @@ public protocol WorkspaceDelegate: AnyObject {
 
     /// The workspace finished downloading all binary artifacts.
     func didDownloadBinaryArtifacts()
+
+    /// Called every time the progress of the git fetch operation updates.
+    func fetchingRepository(from repository: String, objectsFetched: Int, totalObjectsToFetch: Int)
+}
+
+public extension WorkspaceDelegate {
```

Comment:
I think a lot of these were removed so that clients need to implement them.  Should these be added as defaults here in this PR?